### PR TITLE
Cleanup libexpat variables

### PR DESCRIPTION
**Pull request by yadij**
... to use pkg-config compatible naming and take advantage of
squid provided macros better.

Also, add support for pkg-config library auto-detection.

### CODE REVIEW

File: configure.ac
Comment (rousskov) [inline configure.ac:838]: The always-true condition for enabling libexpat does not look right. Please test where libexpat should _not_ be enabled.

**Review by rousskov — CHANGES_REQUESTED** [review]

File: configure.ac
Comment (yadij) [reply configure.ac]: Thanks. Fixed in [1728cd6](https://github.com/squid-cache/squid/pull/1143/commits/1728cd6dd4dc655ba0a0c11eb36ffc09670b9fcc)

**Review by rousskov — APPROVED** [review]
Since this PR now adds a pkg-config check as well, I would adjust the PR title as you did for the libnettle PR.

Disclaimer: I did not review any low-level details.

**Comment by yadij** [comment]
FTR, just waiting until the overlapping libxml changes are merged. Will clear for Anubis after resolving the resulting conflicts.

**Pull request closed without merging**
